CONTRIBUTING.rst

```diff
@@ -322,69 +324,100 @@ The build status for pyiron and all sub packages are given below
 
-   pyiron-0.2.0
+* UpdateDependabotPR.yml: https://github.com/pyiron/pyiron/blob/master/.github/workflows/UpdateDependabotPR.yml
 
-For the automated versioning we use::
+* codeql-analysis.yml: https://github.com/pyiron/pyiron/blob/master/.github/workflows/codeql-analysis.yml
 
-   https://github.com/warner/python-versioneer/
+* deploy.yml: https://github.com/pyiron/pyiron/blob/master/.github/workflows/deploy.yml
 
-So the configuration of the release is included in setup.cfg::
+* docs.yml: https://github.com/pyiron/pyiron/blob/master/.github/workflows/docs.yml
 
-   https://github.com/pyiron/pyiron_base/blob/master/setup.cfg
+* notebooks.yml: https://github.com/pyiron/pyiron_base/blob/master/.github/workflows/notebooks.yml
 
-As the pyiron packages are pure python packages – we use only the Linux Python 3.7 job to build the packages, as defined in the .travis.yml file::
+* pypicheck.yml: https://github.com/pyiron/pyiron_base/blob/master/.github/workflows/pypicheck.yml
 
-   https://github.com/pyiron/pyiron_base/blob/master/.travis.yml
+**UpdateDependabotPR**
 
-The python 3.7 linux tests therefore takes more time, compared to the other tests on travis.
+This workflow is used to keep the dependencies of the setup.py and of the environment.yml synchronized.
+Therefore, it retrieves the head of the branch and runs the ./.ci_support/update_environment.py script to apply the automated changes to the setup.py (by `dependabot`_) also to the environment.yml.
+The script basically checks the title of the PR and bumps the version number from a package to a new version number.
+After that, the version is written into the environment.yml file located at ./.ci_support/environment.yml.
+If this is done successfully, the workflow saves the changes and pushes back to the branch of the PR.
 
-Just like each other commit to the master branch the tagged releases are pushed to pypi.org and anaconda.org::
+**codeql-analysis.yml**
 
-   https://pypi.org/project/pyiron-base/#history
-   https://anaconda.org/pyiron/pyiron_base
+This workflow is used to find vulnerablities inside the codebase with CodeQL.
+First, the head of the branch is retvieved and CodeQL is initialized.
+After that, the CodeQL analysis is performed and the results are returned.
 
-The major difference for pypi (pip) is that tagged releases are the default for pip while installing prerelease versions using pip requires the `--pre` flag.
-`pip install --pre pyiron`
+**deploy.yml**
 
-Those pre-release versions are named `<version_number>.post0.dev<release number>` ::
+This workflow is used to upload and deploy a new release to PyPi. 
+First, the install dependencies in order to create the PyPi distribution.
+After that, the version restriction of the dependencies are lifted to allow for future versions and the PyPi package is build according to the setup.py
+This release is then uploaded to PyPi, but only if it is tagged correctly.
 
-   0.2.0.post0.dev1
+**docs.yml*
```

Comment:
```suggestion
**docs.yml**
```

Reply:
For that I was not sure, if just do not know, until now or if it is outdated.

Are there any other informations have to be changed? Otherwise I would suggest to close this pull request.

Reply:
It is in any case already an improvement! I think it is stated that the pyiron meeting is at 14.00. And there is this 'ASAP as possible' in the text.
I actually would just tell that there is a weekly dev meeting and only tell everyone to get in contact :)

Reply:
Oh, yes. This double ASAP. ... I forgot.
I think related to the pyiron meeting at 14:00, we need also a day. And This meeting is as I know at 15:00 (3 pm german time).
I will edit it later the day.

Reply:
Yes, otherwise everyone not in the team needs to contact one of the organisation in either case. Thus, I would suggest to entirely drop date and time and just say something like 'contact us if you want to participate in the weekly virtual pyiron developer meeting'.

Reply:
Ok, the changes for the doubled ASAP and the pyiron dev meeting are done.

Reply:
Point 3 at [Wait I don't want to read this; ...](https://github.com/pyiron/pyiron/blob/update_contribution_guidelines/CONTRIBUTING.rst#wait-i-dont-want-to-read-this-i-just-have-a-quick-questionbugfix) is mentioned, that people should send us a brief email with their discriptive question. If we do not want to have this kind of questioning via email, we should delete this point or change it to another wording.

Reply:
And I think changing the wording will be the better solution, because otherwise every person, who want to contact us, must have a github account.

Reply:
True, we should mention stackoverflow with tag pyiron - a bug fix would be better on github 😄

Reply:
@niklassiemer In section [pyiron-releases](https://github.com/pyiron/pyiron/blob/update_contribution_guidelines/CONTRIBUTING.rst#pyiron-releases) is a travis.yml file mentioned, which does not exist in the pyiron_base repository. Do you have any recent information about that?

Reply:
Well, recent not really 🤣 https://github.com/pyiron/pyiron/issues/1138 the travis was not free any more. We now have every testing on the github workflows...

Reply:
@niklassiemer Additionally I do not know, how latest the information in the [Debugging](https://github.com/pyiron/pyiron/blob/update_contribution_guidelines/CONTRIBUTING.rst#debugging) are. Did you check this already?

Reply:
This looks OK, there are some nits (e.g. douple \` instead of single \` ) and one should update the syntax in 
```
job = pr.create_job(job_type = pr.job_type.ProtoMD, job_name = 'job')
```
to 
```
job = pr.create.job.ProtoMD('job')
```

Reply:
So it should be better
```
job = pr.create.job.ProtoMD("job")
```
or am I wrong?

Reply:
> Well, recent not really 🤣 #1138 the travis was not free any more. We now have every testing on the github workflows...

Then the information about the pyhton version is also not up to date.
I will change the information about the python version to 3.8, 3.9 and 3.10 on ubuntu and 3.10 for macos and windows.

Reply:
How are the pyiron packages build?
I am not sure.

Reply:
> So it should be better
> 
> ```python
> job = pr.create.job.ProtoMD("job")
> ```
> 
> or am I wrong?

That reads fine. With the double \` I meant the caption with the ``__init__``. However, I should quickly test if it doing the right thing...

Reply:
> How are the pyiron packages build? I am not sure.

https://github.com/pyiron/pyiron_base/blob/master/.github/workflows/deploy.yml should build the packages (and deploy them if it is a release).

Reply:
Ok, now I checked I edited a different file, where the information about the travis.yml was still in.

GitHub is confusing me. I hope that will reduce in some time.